freeipa/src/main/java/com/sequenceiq/freeipa/service/operation/FlowOperationService.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.freeipa.service.operation;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.flow.api.model.operation.OperationFlowsView;
+import com.sequenceiq.flow.api.model.operation.OperationResource;
+import com.sequenceiq.flow.api.model.operation.OperationView;
+import com.sequenceiq.flow.converter.OperationDetailsPopulator;
+import com.sequenceiq.flow.service.FlowService;
+import com.sequenceiq.freeipa.flow.freeipa.provision.FreeIpaProvisionFlowConfig;
+import com.sequenceiq.freeipa.flow.stack.provision.StackProvisionFlowConfig;
+
+@Service
+public class FlowOperationService {
+
+    @Inject
+    private FlowService flowService;
+
+    @Inject
+    private OperationDetailsPopulator operationDetailsPopulator;
+
+    public OperationView getOperationProgressByEnvironmentCrn(String resourceCrn, boolean detailed) {
```

Comment:
If this is really always for looking up by environment CRN (and not freeipa CRN, for example), perhaps rename `resourceCrn` to `environmentCrn`.